tfhe/src/shortint/ciphertext/mod.rs

```diff
@@ -83,6 +123,7 @@ impl Degree {
 pub struct Ciphertext {
     pub ct: LweCiphertextOwned<u64>,
     pub degree: Degree,
+    pub noise_level: NoiseLevel,
```

Comment:
So as we know adding this field breaks deserialization

Do we plan to do something to mitigate that ?

Reply:
with the 0.5.0 we will do the semver trick, we will most likely provide a way to load old ciphertext and set the noise to the max for that ciphertext or do a bootstrapping right away I would say

for 0.5.0 I would try to break as many things as possible

Reply:
@mayeul-zama didn't we say we want the noise level to be private to be able to crash with a given feature if the noise level exceeds a valid noise level ?

Reply:
I don't remember that
I don't see how it's needed to have this crash feature

Reply:
how do you guarantee we never go over the limit without a setter ?

Reply:
I was thinking of adding the assert in the few places where it's updated
But indeed your proposition is safer
I'll implement it

Reply:
The setter will likely need an input to indicate the max noise level, which will require an input from the Server Key most likely, the extra input would be ignored if we are not checking (i.e. if the noise assert feature is disabled) but used otherwise

once we have that I believe most pieces of the puzzle will be ready

Reply:
there remains a question for the maximum level of noise that should be stored in parameters

Reply:
or each ciphertext brings it's own max noise level which may be a bit wasteful

Reply:
Maybe putting an assert on the noise before bootstrap might be enough to catch errors since bootstraps are very frequent and the whole norm2 is computed around that ?

Reply:
that's a very good point

Reply:
it might make identifying the faulty operation harder though if you have a chain of adds, so... not sure I like it much

Reply:
Yes, and if you too many linear operations without a PBS after, the assert won't be triggered

Reply:
Yeah its less precise and uses the fact that in practive you are very limited in the number of leveled operation you can do

Reply:
But making it private forces the use of a constructor instead of a normal struct construction which is less ergonomic IMO

Reply:
yes but that's a very small price to pay, even though it crossed my mind